packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -58,6 +59,96 @@ const {
 const { fakeFileFactory, fakeGranuleFactoryV2 } = require('../../../lib/testUtils');
 const Granule = require('../../../models/granules');
 
+/**
+ * Helper function for updating an existing granule with a static payload and validating
+ *
+ * @param {Object} t -- Used for the test context
+ * @param {boolean} writeFromMessage -- Calls writeGranulesFromMessage function if true,
+ *   writeGranuleFromApi otherwise
+ * @returns {Object} -- Updated granule objects from each datastore and PG-translated payload
+ *   updatedPgGranuleFields,
+ *   pgGranule,
+ *   esGranule,
+ *   dynamoGranule,
+ **/
+const updateGranule = async (t, writeFromMessage = false) => {
+  const {
+    collectionCumulusId,
+    esClient,
+    esGranulesClient,
+    executionCumulusId,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    providerCumulusId,
+    knex,
+  } = t.context;
+
+  // Update existing granule with a partial granule object
+  const updateGranulePayload = {
+    granuleId,
+    collectionId: granule.collectionId,
+    cmrLink: 'updatedGranuled.com', // Only field we're changing
+    // FUTURE: In order to update a granule, the payload must include status and
+    // the status must be 'completed' or 'failed'
+    // if it's running or queued, it will try to insert the granule, not upsert
+    status: granule.status,
+  };
```

Comment:
TODO:

Test a payload with e.g. `files: null` we would expect it to remove all files from both ES/Dynamo and PG